apps/builder/app/builder/features/sidebar-left/panels/components/components.tsx

```diff
@@ -1,160 +1,55 @@
-import { useState } from "react";
-import { useStore } from "@nanostores/react";
-import { createPortal } from "react-dom";
-import type { Instance } from "@webstudio-is/project-build";
 import {
   type ComponentName,
   getComponentMeta,
   getComponentNames,
+  WsComponentMeta,
+  componentCategories,
 } from "@webstudio-is/react-sdk";
 import {
   theme,
   Flex,
-  useDrag,
-  type Point,
   ComponentCard,
+  Tooltip,
 } from "@webstudio-is/design-system";
 import { PlusIcon } from "@webstudio-is/icons";
-import { findClosestDroppableTarget } from "~/shared/tree-utils";
-import {
-  instancesStore,
-  selectedInstanceSelectorStore,
-  selectedPageStore,
-} from "~/shared/nano-states";
-import { useSubscribe, type Publish } from "~/shared/pubsub";
-import {
-  isCanvasPointerEventsEnabledStore,
-  useCanvasRect,
-} from "~/builder/shared/nano-states";
-import { insertNewComponentInstance } from "~/shared/instance-utils";
-import { zoomStore } from "~/shared/nano-states/breakpoints";
+import type { Publish } from "~/shared/pubsub";
 import type { TabName } from "../../types";
 import { Header, CloseButton } from "../../header";
+import { ArrowFocus } from "@webstudio-is/design-system";
+import { CollapsibleSection } from "~/builder/shared/collapsible-section";
+import { useDraggable } from "./use-draggable";
 
-const DragLayer = ({
-  component,
-  point,
-}: {
-  component: Instance["component"];
-  point: Point;
-}) => {
-  const meta = getComponentMeta(component);
-  if (meta === undefined) {
-    return null;
-  }
+const metasByComponentName: Map<ComponentName, WsComponentMeta> = new Map();
+const metasByCategory: Map<
+  WsComponentMeta["category"],
+  Array<WsComponentMeta>
+> = new Map();
+const componentNamesByMeta: Map<WsComponentMeta, ComponentName> = new Map();
 
-  return createPortal(
-    <Flex
-      // Container is used to position card
-      css={{
-        position: "absolute",
-        inset: 0,
-      }}
-    >
-      <ComponentCard
-        label={meta.label}
-        icon={<meta.Icon />}
-        style={{
-          transform: `translate3d(${point.x}px, ${point.y}px, 0)`,
-        }}
-      />
-    </Flex>,
-    document.body
-  );
-};
+for (const name of getComponentNames()) {
```

Comment:
This top level stuff may not play well with injecting components. Might be better to do inside component